### PR TITLE
Remove iRODS dependencies in cmake and build hook

### PR DESCRIPTION
**Pull request by alanking**
This change removes unnecessary iRODS dependencies in CMakeLists.txt for mungefs, though many steps from the iRODS CMakeLists.txt were borrowed. Not all things may be required in this version of the file, but there is no more dependency on an irods-dev or runtime environment when building mungefs.

Now that mungefs does not need certain iRODS definitions for cmake, irods dev/runtime are no longer needed to build/install/run mungefs. Added things to build hook being implicitly supplied by irods dev
and runtime packages.

See [CMakeLists.txt in irods/irods](https://github.com/irods/irods/blob/master/CMakeLists.txt).

### CODE REVIEW

**Comment by alanking** [comment]
This experienced issues on CentOS 7, so more work will be needed. Do not merge yet.

File: CMakeLists.txt [hunk 1]
```diff
@@ -52,6 +164,7 @@ target_link_libraries(
   ${IRODS_EXTERNALS_FULLPATH_BOOST}/lib/libboost_filesystem.so
   ${IRODS_EXTERNALS_FULLPATH_BOOST}/lib/libboost_system.so
   ${IRODS_EXTERNALS_FULLPATH_ARCHIVE}/lib/libarchive.so
+  ${IRODS_EXTERNALS_FULLPATH_CLANG_RUNTIME}/lib/libc++abi.so.1
```
Comment (alanking) [inline CMakeLists.txt]: This was needed to help Cen7 find the SO at runtime. I'm guessing the link_libraries call on **line 144** is supposed to take care of this, but apparently that doesn't work on CentOS (although it does for Ubuntu).

**Comment by trel** [comment]
probably just due to a lack of symlink from .so->.so.1 in that /lib directory.

**Comment by alanking** [comment]
~~I suspect that the issue seen on Cen7 was due to using an incorrect SHA when building mungefs in CI. Everything seems okay now. Thanks!~~

Sorry, still seem to be having issues on Cen7.

**Comment by trel** [comment]
awaiting one more sanity check... through CI.

**Comment by alanking** [comment]
This will need to run through CI before merging, but the issue with Cen7 has been resolved. Will notify when ready.

**Comment by alanking** [comment]
Cen7 passed in CI using mungefs, so we are good to merge.

**Pull request merged**